core/xml.js

```diff
@@ -783,7 +783,8 @@ Blockly.Xml.domToFieldVariable_ = function(workspace, xml, text, field) {
     type = '';
   }
 
-  var variable = Blockly.Variables.getOrCreateVariablePackage(workspace, xml.id,
+  // pxt-blockly: variable ID and variable name are both unique, only use name
+  var variable = Blockly.Variables.getOrCreateVariablePackage(workspace, null,
```

Comment:
can the variable name be empty?

Reply:
I just tried it on micro:bit and it doesn't seem like it